### PR TITLE
[FIX] fix transform_cigar_right_into_insertion if ref_pos == until.

### LINKED ISSUE #23: Segmentation fault for few SVs
Hi,

I am working on PacBio data and want to use sviper in my pipeline. I used the latest version and successfully corrected most of my SVs. However, I ran into segmentation fault for very few SVs. No additional error information was raised and sviper just stop without output vcf.

Please see the following link for a reproducible example and log file: https://www.dropbox.com/s/mhntmdae9px243g/sviper.zip?dl=0

The code I used is:
sviper -s short_read.bam -l long_read.bam -r hg38.fa -c test_input.vcf -o test_output -t 1

Thanks,
Han

### PR DESCRIPTION
**Pull request by smehringer**
fixes #23

### CODE REVIEW

File: include/sviper/merge_split_alignments.h [hunk 1]
```diff
@@ -114,6 +114,15 @@ inline void truncate_cigar_right(seqan::BamAlignmentRecord & record, int until)
...
 
+// we have a supp record (`record`) that starts before the primary record (begin position `until`)
+// e.g.      XXXXXXXXXXXXXX
+//                  YYYYYYYYYYYYYYYYYYYYYY
+// We want to change the cigar of `record` in a way that everything that overlaps with the primary record is put
+// into a big insertion:
+// e.g.      XXXXXXXI
+//                  YYYYYYYYYYYYYYYYYYYYYY
+// s.t. both cigar strings can be merged into one merged alignment:
+// e.g.      XXXXXXXIYYYYYYYYYYYYYYYYYYYYYY
```
Comment (Irallia) [inline include/sviper/merge_split_alignments.h:125]: So just to check if I understand it correct:
You have 2 cigar strings X and Y. X from record, Y from primary record.
eg. record cigar: 3D4M5D2I7M with pos 10 before the primary one. 
You add a big insertion after pos 10 -> 3D4M3D(0+2+7)I and than you can merge this cigar with the primary one? (You only add M's and I's right?)
and `until` is the 10 here?

File: include/sviper/merge_split_alignments.h [hunk 2]
```diff
@@ -125,19 +134,11 @@ inline void transform_cigar_right_into_insertion(seqan::BamAlignmentRecord & rec
...
+    SEQAN_ASSERT_NEQ(cigar_pos, 0u); // no advancement would have happened. edge case: until=1?
     --cigar_pos; // move to cigar operation that over stepped
     unsigned insertion_size{0};
 
-    if (ref_pos == until)
-    {
-        --cigar_pos;
-    }
+    // if (ref_pos == until) we don't need to do anything
     if (ref_pos > until)
```
Comment (Irallia) [inline include/sviper/merge_split_alignments.h:142]: If `ref_pos > until` we have an overlap otherwise not?

**Review by Irallia — CHANGES_REQUESTED** [review]
only questions of understanding....

by the way, you dont handle `X` and `=` in sviper right?
Used instead of M:
'=' means REF and READ are the same.
'X' means REF and READ are not the same.

File: include/sviper/merge_split_alignments.h (hunk 1 above)
Comment (smehringer) [reply include/sviper/merge_split_alignments.h:125]: yes exactly :)

File: include/sviper/merge_split_alignments.h (hunk 2 above)
Comment (smehringer) [reply include/sviper/merge_split_alignments.h:142]: yes! :)

**Comment by smehringer** [comment]
> by the way, you dont handle X and = in sviper right?

no currently only simple cigar strings are allowed

**Review by Irallia — APPROVED** [review]

**Pull request merged**